Tests/Test_SDL.cpp

```diff
@@ -0,0 +1,228 @@
+            case SDL::e_VarCreatable: break;                                                    \
+            case SDL::e_VarDouble: CHECK_VALUE(varA, varB, m_double) break;                     \
+            case SDL::e_VarTime: CHECK_VALUE(varA, varB, m_time) break;                         \
+            case SDL::e_VarByte: CHECK_VALUE(varA, varB, m_byte) break;                         \
+            case SDL::e_VarShort: CHECK_VALUE(varA, varB, m_short) break;                       \
+            /* Skipping e_VarAgeTimeofDay (no value) */                                         \
+            case SDL::e_VarAgeTimeOfDay: break;                                                 \
+            case SDL::e_VarVector3:                                                             \
+            case SDL::e_VarPoint3:                                                              \
+                CHECK_VALUE(varA, varB, m_vector)                                               \
+                break;                                                                          \
+            case SDL::e_VarQuaternion: CHECK_VALUE(varA, varB, m_quat) break;                   \
+            case SDL::e_VarRgb:                                                                 \
+            case SDL::e_VarRgba:                                                                \
+                CHECK_VALUE(varA, varB, m_color)                                                \
+                break;                                                                          \
+            case SDL::e_VarRgb8:                                                                \
+            case SDL::e_VarRgba8:                                                               \
+                CHECK_VALUE(varA, varB, m_color8)                                               \
+                break;                                                                          \
+            /* Skipping e_VarStateDesc for now... */                                            \
+            case SDL::e_VarStateDesc: break;                                                    \
+            default: CHECK(0); break;                                                           \
+        }                                                                                       \
+}
+
+static SDL::State CreateState()
+{
+    SDL::StateDescriptor* desc = SDL::DescriptorDb::FindDescriptor("Test", 1);
+    REQUIRE(desc != nullptr);
+
+    auto boolVarIdx = desc->m_varmap.find("bTestVar1");
+    auto intVarIdx = desc->m_varmap.find("iTestVar3");
+
+    REQUIRE(boolVarIdx != desc->m_varmap.end());
+    REQUIRE(intVarIdx != desc->m_varmap.end());
+
+    SDL::State state(desc);
+    state.data()->m_vars[boolVarIdx->second].data()->m_bool = new bool[1] { true };
+    state.data()->m_vars[intVarIdx->second].data()->m_int = new int32_t[1] { 6 };
```

Comment:
These are leaks as written (the storage is already allocated):

```suggestion
    state.data()->m_vars[boolVarIdx->second].data()->m_bool[0] = true;
    state.data()->m_vars[intVarIdx->second].data()->m_int[0] = 6;
```